packages/vite/src/node/server/index.ts

```diff
@@ -720,16 +713,17 @@ export function resolveServerOptions(
   let allowDirs = server.fs?.allow
 
   if (!allowDirs) {
-    allowDirs = [server.fs?.root || searchForWorkspaceRoot(root)]
-  }
-  allowDirs = allowDirs.map((i) => resolvedAllowDir(root, i))
+    allowDirs = [resolvedAllowDir(root, searchForWorkspaceRoot(root))]
 
-  // only push client dir when vite itself is outside-of-root
-  const resolvedClientDir = resolvedAllowDir(root, CLIENT_DIR)
-  if (!allowDirs.some((i) => resolvedClientDir.startsWith(i))) {
-    allowDirs.push(resolvedClientDir)
+    // only push client dir when vite itself is outside-of-root
+    const resolvedClientDir = resolvedAllowDir(root, CLIENT_DIR)
+    if (!allowDirs.some((i) => resolvedClientDir.startsWith(i))) {
```

Comment:
Something that changed with this PR is that the vite client dir is no longer added if there are customs allow dirs. This means that yarn link vite will not work in setups that require allow like integrations with rails. Was this intentional?

Reply:
Yeah, I was, as I kinda think we should give user full control of this options (no implicit append). But after a second thought, it might be not that important as the client dir is rather a small scope and user could use `denied` to exclude them if they really don't want it in the future (which does not make sense tho)